source/performance/F0.Minesweeper.Logic.Benchmarks/LocationShuffler/LocationShufflerBenchmarks.cs

```diff
@@ -0,0 +1,69 @@
+using System.Diagnostics;
+using F0.Minesweeper.Logic.Abstractions;
+using F0.Minesweeper.Logic.LocationShuffler;
+
+namespace F0.Minesweeper.Logic.Benchmarks.LocationShuffler
+{
+	public class LocationShufflerBenchmarks
+	{
+		private readonly ILocationShuffler globallyUniqueIdentifier = new GuidLocationShuffler();
+		private readonly ILocationShuffler randomOrder = new RandomOrderLocationShuffler();
+		private readonly ILocationShuffler fisherYates = new FisherYatesLocationShuffler();
+
+		private IEnumerable<Location> minefield = null!;
+
+		[ParamsSource(nameof(Parameters))]
+		public Param Parameter { get; set; } = null!;
+
+		[GlobalSetup]
+		public void GlobalSetup()
+		{
+			uint size = Parameter.Width * Parameter.Height;
+			List<Location> locations = new((int)size);
+
+			for (uint width = 0; width < Parameter.Width; width++)
+			{
+				for (uint height = 0; height < Parameter.Height; height++)
+				{
+					locations.Add(new Location(width, height));
+				}
+			}
+
+			Debug.Assert(locations.Count == size);
+			minefield = locations;
+		}
+
+		[Benchmark]
+		public IReadOnlyCollection<Location> GloballyUniqueIdentifier()
+			=> globallyUniqueIdentifier.ShuffleAndTake(minefield, (int)Parameter.MineCount);
+
+		[Benchmark]
+		public IReadOnlyCollection<Location> RandomOrder()
+			=> randomOrder.ShuffleAndTake(minefield, (int)Parameter.MineCount);
+
+		[Benchmark]
+		public IReadOnlyCollection<Location> FisherYates()
+			=> fisherYates.ShuffleAndTake(minefield, (int)Parameter.MineCount);
+
+		public static IEnumerable<Param> Parameters()
+		{
+			MinefieldOptions[] collection = new[]
+			{
+				MinefieldOptions.Easy,
+				MinefieldOptions.Medium,
+				MinefieldOptions.Hard,
+			};
+
+			foreach (MinefieldOptions item in collection)
+			{
+				yield return new Param(item.Width, item.Height, item.MineCount);
```

Comment:
Due to the nature of the `yield return` I just want to ask one thing in case you know this: are methods tagged with the `[Benchmark]` attribute only called when all setup stuff like `ParamsSource` and `GlobalSetup` are fully finished? Or is it possible that something like the `yield return` during setup could interfere with benchmarking results?

Reply:
The `c'tor`, `[GlobalSetup]` and `[GlobalCleanup]` are not part of the `[Benchmark]`'s statistics and heuristics.